src/components/Modal/ModalAddLink.js

```diff
@@ -0,0 +1,166 @@
+    flex-direction: column;
+    gap: 24px;
+
+    .addlink-text {
+      padding: 0;
+      gap: 8px;
+      width: 100%;
+
+      h1 {
+        color: #373740;
+        font-size: 1.25rem;
+        font-weight: 700;
+      }
+
+      p {
+        color: #9fa6b2;
+        font-size: 0.875rem;
+        font-weight: 400;
+        line-height: 22px;
+      }
+
+      img {
+        position: absolute;
+        top: -20px;
+        right: -25px;
+        cursor: pointer;
+      }
+    }
+
+    .addlink-folder {
+      padding: 0;
+      gap: 4px;
+      width: 100%;
+      overflow: scroll;
+      overflow-x: hidden;
+
+      div {
+        padding: 8px;
+        flex-direction: row;
+        align-items: center;
+        gap: 8px;
+        border-radius: 8px;
+        width: 100%;
+
+        h2 {
+          color: #373740;
+          font-size: 1rem;
+          font-weight: 400;
+          line-height: 24px;
+        }
+
+        h3 {
+          color: #9fa6b2;
+          font-size: 0.875rem;
+          font-weight: 400;
+        }
+
+        img {
+          position: absolute;
+          right: 8px;
+          display: none;
+        }
+      }
+
+      div:hover {
+        background-color: #f0f6ff;
+      }
+
+      .selected {
+        background-color: #f0f6ff;
+
+        img {
+          display: inherit;
+        }
+      }
+    }
+
+    button {
+      width: 280px;
+      padding: 16px 20px;
+      border-radius: 8px;
+      background: linear-gradient(91deg, #6d6afe 0.12%, #6ae3fe 101.84%);
+      border: 0px;
+      cursor: pointer;
+
+      color: #f5f5f5;
+      font-size: 1rem;
+      font-style: normal;
+      font-weight: 600;
+      line-height: normal;
+    }
+  }
+`;
+
+const ModalAddLink = ({ handleClose, handleButton, link, buttonInfo }) => {
+  const [selectedFolders, setSelectedFolders] = useState([]);
+
+  const handleFolderSelected = (folderId) => {
+    if (selectedFolders.includes(folderId)) {
+      setSelectedFolders(selectedFolders.filter((id) => id !== folderId));
+    } else {
+      setSelectedFolders([...selectedFolders, folderId]);
+    }
+  };
+
+  return (
+    <ModalContainer onClick={(e) => e.preventDefault()}>
+      <div>
+        <div className="addlink-text">
+          <h1>폴더에 추가</h1>
+          <p>{link}</p>
+          <img src={close} alt="close" onClick={handleClose} />
+        </div>
+
+        <div className="addlink-folder">
+          {buttonInfo
+            ? buttonInfo.map((item) => (
+                <div
+                  key={item.id}
+                  onClick={() => handleFolderSelected(item.id)}
+                  className={
+                    selectedFolders.includes(item.id) ? "selected" : undefined
+                  }
+                >
+                  <h2>{item.name}</h2>
+                  <h3>{item.link.count}개 링크</h3>
+                  <img src={check} alt="check" />
+                </div>
+              ))
+            : alert("폴더정보는 8주차 과제(folder페이지)부터 제공됩니다.")}
```

Comment:
```suggestion
  useEffect(() => {
    if (!buttonInfo) {
      alert("폴더정보는 8주차 과제(folder페이지)부터 제공됩니다.");
    }
  }, [])
  .
  .
  .

          {buttonInfo
            && buttonInfo.map((item) => (
                <div
                  key={item.id}
                  onClick={() => handleFolderSelected(item.id)}
                  className={
                    selectedFolders.includes(item.id) ? "selected" : undefined
                  }
                >
                  <h2>{item.name}</h2>
                  <h3>{item.link.count}개 링크</h3>
                  <img src={check} alt="check" />
                </div>
              ))
```
[중요도: 높음] 이런 형식을 추천드립니다.
props라서 당장에 문제가 되지는 않지만 원래 하던 방식으로 진행하게 된다면 리렌더링 될 때 마다 저 alert가 노출될 수 있습니다!

그래서 저는 컴포넌트 return문 부터에는 순수한 함수를 렌더하게 하는 것은 안쓰고 있으니 바꿔보시는 것도 좋아보입니다!